frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/ReverificationResultHandler.java

```diff
@@ -4,15 +4,130 @@
+import static uk.gov.di.authentication.frontendapi.domain.FrontendAuditableEvent.REVERIFY_SUCCESSFUL_VERIFICATION_INFO_RECEIVED;
+import static uk.gov.di.authentication.frontendapi.domain.FrontendAuditableEvent.REVERIFY_UNSUCCESSFUL_TOKEN_RECEIVED;
+import static uk.gov.di.authentication.shared.entity.ErrorResponse.ERROR_1058;
+import static uk.gov.di.authentication.shared.entity.ErrorResponse.ERROR_1059;
+import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyErrorResponse;
 import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyResponse;
 
-public class ReverificationResultHandler
+public class ReverificationResultHandler extends BaseFrontendHandler<ReverificationResultRequest>
         implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    private static final Logger LOG = LogManager.getLogger(ReverificationResultHandler.class);
+    private final ReverificationResultService reverificationResultService;
+    private final AuditService auditService;
+
+    public ReverificationResultHandler(
+            ConfigurationService configurationService,
+            SessionService sessionService,
+            ClientSessionService clientSessionService,
+            ClientService clientService,
+            AuthenticationService authenticationService,
+            ReverificationResultService reverificationResultService,
+            AuditService auditService) {
+        super(
+                ReverificationResultRequest.class,
+                configurationService,
+                sessionService,
+                clientSessionService,
+                clientService,
+                authenticationService);
+        this.reverificationResultService = reverificationResultService;
+        this.auditService = auditService;
+    }
+
+    public ReverificationResultHandler() {
+        this(ConfigurationService.getInstance());
+    }
+
+    public ReverificationResultHandler(ConfigurationService configurationService) {
+        super(ReverificationResultRequest.class, configurationService, true);
+        this.reverificationResultService = new ReverificationResultService(configurationService);
+        this.auditService = new AuditService(configurationService);
+    }
 
     @Override
     public APIGatewayProxyResponseEvent handleRequest(
             APIGatewayProxyRequestEvent input, Context context) {
-        return generateApiGatewayProxyResponse(200, "hello world");
+        return super.handleRequest(input, context);
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequestWithUserContext(
+            APIGatewayProxyRequestEvent input,
+            Context context,
+            ReverificationResultRequest request,
+            UserContext userContext) {
+
+        AuditContext auditContext =
+                new AuditContext(
+                        userContext.getClientId(),
```

Comment:
Here you can use the constructor where you pass in the user context, which does the first three fields, and the last field of this for free.